### PR TITLE
Trace until condition

### PR DESCRIPTION
**Pull request by torusrxxx**
This change is [Reviewable](https://reviewable.io/reviews/x64dbg/x64dbg/749)

### CODE REVIEW

**Comment by mrexodia** [comment]
Maybe we should set a max step limit here?

File: src/dbg/debugger.cpp [hunk 1]
```diff
@@ -774,6 +775,48 @@ void cbRtrStep()
         StepOver((void*)cbRtrStep);
 }
 
+void cbTOCNDStep()
+{
+    duint value = 1;
+    if (!RtcondCondition || RtcondCondition->Calculate(value, valuesignedcalc()) == false)
+    {
+        delete RtcondCondition;
```
Comment (mrexodia) [inline src/dbg/debugger.cpp]: This will crash if the condition is null

File: src/dbg/debugger_commands.cpp [hunk 2]
```diff
@@ -1255,6 +1255,52 @@ CMDRESULT cbDebugeRtr(int argc, char* argv[])
...
+    {
+        dputs("Too many arguments.");
+        return STATUS_ERROR;
+    }
+    else if (argc < 1)
+    {
+        dputs("Too few arguments.");
+        return STATUS_ERROR;
+    }
+    if (RtcondCondition != nullptr)
```
Comment (mrexodia) [inline src/dbg/debugger_commands.cpp]: This might need locking, gotta check it out.

File: src/dbg/debugger.cpp (hunk 1 above)
Comment (justanotheranonymoususer) [inline src/dbg/debugger.cpp]: http://stackoverflow.com/questions/4190703

**Comment by mrexodia** [comment]
Thanks a lot!

**Pull request merged**